wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheMonitor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.cache;
+
+/**
+ * Interface for cache health monitors. This class uses a background thread to monitor cache
+ * cluster health and manage state transitions between HEALTHY, SUSPECT, and DEGRADED states.
+ */
+public interface CacheMonitor extends Runnable {
```

Comment:
You don't need to make this an interface if you only have 1 implementation of it. Just use a class here. 

You can actually declare lots of the methods in this class to be `static protected`. That way `CacheConnection` doesn't need to have a handle on this `CacheMonitor`